api/src/resources/comment.rs

```diff
@@ -36,6 +36,14 @@ impl FromStr for Id {
 #[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash)]
 pub struct Uid(pub String);
 
+impl FromStr for Uid {
+    type Err = Error;
```

Comment:
Why do you need to declare this type here? I'm just struggling to see where this is used

Reply:
This `type` is part of the `FromStr` trait that we're implementing. It is required that we specify what the error type is that this trait returns, which in this case is our global error type, `Error` (the return type `Result<Self>` is an alias for `std::result::Result<Self, Error>`).

More generally, generic traits can have associated types that are named, rather than positional. So in this case, the trait definition is:

```rust
pub trait FromStr {
    type Err;  // <- says the trait implemention must specify this type
    fn from_str(s: &str) -> Result<Self, Self::Err>;
}
```

which you could have defined using positional rather than named associated types, something like:

```rust
pub trait FromStr<Err> {
    fn from_str(s: &str) -> Result<Self, Err>;
}
```

https://doc.rust-lang.org/std/str/trait.FromStr.html#associated-types